### PR TITLE
[docs] Restructure how-to section to make it more readable

### PR DESCRIPTION
**Pull request by wjayesh**
## Describe changes
- I restructured the existing how-to section based on this idea: https://www.notion.so/zenml/Restructure-how-to-section-after-we-write-all-other-docs-to-make-it-easier-to-read-11ff8dff253880e9883bcd9567131689
- Added redirects so older links point to new locations

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: .gitbook.yaml [hunk 1]
```diff
@@ -17,3 +17,108 @@ redirects:
...
+  how-to/visualize-artifacts/visualizations-in-dashboard.md: how-to/data-artifact-management/visualize-artifacts/visualizations-in-dashboard.md
+  how-to/visualize-artifacts/disabling-visualizations.md: how-to/data-artifact-management/visualize-artifacts/disabling-visualizations.md
+
+  # Model Management redirects
+  how-to/use-the-model-control-plane/README.md: how-to/model-management-metrics/model-control-plane/README.md
+  how-to/use-the-model-control-plane/register-a-model.md: how-to/model-management-metrics/model-control-plane/register-a-model.md
+  how-to/use-the-model-control-plane/delete-a-model.md: how-to/model-management-metrics/model-control-plane/delete-a-model.md
+  how-to/use-the-model-control-plane/associate-a-pipeline-with-a-model.md: how-to/model-management-metrics/model-control-plane/associate-a-pipeline-with-a-model.md
+  how-to/use-the-model-control-plane/connecting-artifacts-via-a-model.md: how-to/model-management-metrics/model-control-plane/connecting-artifacts-via-a-model.md
+  how-to/use-the-model-control-plane/model-versions.md: how-to/model-management-metrics/model-control-plane/model-versions.md
```
Comment (htahir1) [inline .gitbook.yaml:103]: left side should not end in .md

**Review by htahir1 — CHANGES_REQUESTED** [review]
You moved all these but did not change the links within the content! :-D

File: .gitbook.yaml (hunk 1 above)
Comment (wjayesh) [reply .gitbook.yaml:103]: yes, fixing it!

**Comment by wjayesh** [comment]
@htahir1 we don't need to change the links in all files, that's what the redirects help with, correct? Just verified that links in pages that point to the moved docs still work. For example, check out the `this docs page` link in the deployed docs https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/-MCl0MaN3pJfPa2vq2Rb/~/revisions/79v2nDlm4bhHjIxCY4h5/stack-components/orchestrators/airflow#additional-configuration

**Comment by htahir1** [comment]
@wjayesh i know but isnt it weird to have intentionally broken links? can we just change these to point to the right paths please?

File: docs/book/how-to/advanced-topics/manage-zenml-server/best-practices-upgrading-zenml.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ loaded_artifact = artifact.load()
 While ZenML strives for backward compatibility, be prepared for occasional breaking changes (e.g., [the Pydantic 2 upgrade](https://github.com/zenml-io/zenml/releases/tag/0.60.0)).
 
 - **Changelog Review**: Always review the [changelog from new releases](https://github.com/zenml-io/zenml/releases) for new syntax, instructions, or breaking changes.
-- **Migration Scripts**: Use provided [migration scripts](../../how-to/manage-the-zenml-server/migration-guide/migration-guide.md) when available to handle database schema changes.
+- **Migration Scripts**: Use provided [migration scripts](migration-guide/migration-guide.md) when available to handle database schema changes.
```
Comment (htahir1) [inline docs/book/how-to/advanced-topics/manage-zenml-server/best-practices-upgrading-zenml.md:70]: is this correct?

**Review by htahir1 — CHANGES_REQUESTED** [review]
one small thing but totherwise looks good to go

File: docs/book/how-to/advanced-topics/manage-zenml-server/best-practices-upgrading-zenml.md (hunk 2 above)
Comment (wjayesh) [reply docs/book/how-to/advanced-topics/manage-zenml-server/best-practices-upgrading-zenml.md:70]: yes, the migration guide is in the same directory and resolves to the right file

File: .gitbook.yaml [hunk 3]
```diff
@@ -15,6 +15,161 @@ redirects:
...
   getting-started/deploying-zenml/manage-the-deployed-services/troubleshoot-your-deployed-server: how-to/manage-the-zenml-server/troubleshoot-your-deployed-server.md
   how-to/stack-deployment/implement-a-custom-integration: how-to/contribute-to-zenml/implement-a-custom-integration.md
-  how-to/setting-up-a-project-repository/best-practices: how-to/setting-up-a-project-repository/set-up-repository.md
+  how-to/setting-up-a-project-repository/best-practices: how-to/project-setup-and-management/setting-up-a-project-repository/set-up-repository.md
   getting-started/zenml-pro/system-architectures: getting-started/system-architectures.md
-  how-to/build-pipelines/name-your-pipeline-and-runs: how-to/build-pipelines/name-your-pipeline-runs.md
+  how-to/build-pipelines/name-your-pipeline-and-runs: how-to/pipeline-development/build-pipelines/name-your-pipeline-runs.md
+
+  # Project Setup redirects
+  how-to/setting-up-a-project-repository/README: how-to/project-setup-and-management/setting-up-a-project-repository/README.md
```
[the remark this answers is not in the document]
Comment (wjayesh) [reply .gitbook.yaml:23]: ah, i see. let me update it

[the remark this answers is not in the document]
Comment (wjayesh) [reply .gitbook.yaml]: ok done and tested for all README files

**Review by htahir1 — APPROVED** [review]

**Pull request merged**